cms/static/js/views/modals/course_outline_modals.js

```diff
@@ -590,47 +590,65 @@ define(['jquery', 'backbone', 'underscore', 'gettext', 'js/views/baseview',
         }
     });
 
-    StaffLockEditor = AbstractEditor.extend({
-        templateName: 'staff-lock-editor',
-        className: 'edit-staff-lock',
-        isModelLocked: function() {
-            return this.model.get('has_explicit_staff_lock');
+    ContentVisibilityEditor = AbstractEditor.extend({
+        templateName: 'content-visibility-editor',
+        className: 'edit-content-visibility',
+        modelVisibility: function() {
+            if (this.model.get('has_explicit_staff_lock')) {
+                return 'staff_only';
+            }
+            else if (this.model.get('hide_after_due')) {
+                return 'hide_after_due';
+            }
+            /* TODO: uncomment when hide_after_submit is added
+            else if (this.model.get('hide_after_submit')) {
+                return 'hide_after_submit';
+            }
+            */
+            else {
+                return 'visible'
+            }
         },
 
-        isAncestorLocked: function() {
-            return this.model.get('ancestor_has_staff_lock');
+        ancestorVisibility: function() {
+            return this.model.get('ancestor_visibility');
         },
 
         afterRender: function () {
             AbstractEditor.prototype.afterRender.call(this);
-            this.setLock(this.isModelLocked());
+            this.setVisibility(this.modelVisibility());
         },
 
-        setLock: function(value) {
-            this.$('#staff_lock').prop('checked', value);
+        setVisibility: function(value) {
+            // TODO: set a radio button here
         },
 
-        isLocked: function() {
-            return this.$('#staff_lock').is(':checked');
+        currentVisibility: function() {
+            // TODO: return currently selected radio option
         },
 
         hasChanges: function() {
-            return this.isModelLocked() !== this.isLocked();
+            return this.modelVisibility() !== this.currentVisibility();
         },
 
         getRequestData: function() {
-            return this.hasChanges() ? {
-                publish: 'republish',
-                metadata: {
-                    visible_to_staff_only: this.isLocked() ? true : null
+            currentVisibility = this.currentVisibility()
+            if (this.hasChanges) {
+                return {
+                    publish: 'republish',
+                    metadata: {
+                        visible_to_staff_only: currentVisibility == 'staff_only' ? true : null,
+                        content_visibility: ['hide_after_due', 'hide_after_submit'].indexOf(currentVisibility) ? currentVisibility : null
```

Comment:
@jcdyer FYI - I believe this is where I'll be passing xblock field data into the new backend.